modules/charon-core/src/main/java/org/wso2/charon3/core/objects/RoleV2.java

```diff
@@ -0,0 +1,621 @@
+            this.attributeList.put(SCIMConstants.RoleSchemaConstants.DISPLAY_NAME, displayAttribute);
+        }
+    }
+
+    /**
+     * Deletes the current value of display name and rename it with the given value.
+     *
+     * @param displayName New display name.
+     */
+    public void replaceDisplayName(String displayName) {
+
+        replaceSimpleAttribute(SCIMSchemaDefinitions.SCIMRoleSchemaDefinition.DISPLAY_NAME, displayName);
+    }
+
+    /**
+     * Get the users of the role.
+     *
+     * @return List of users.
+     */
+    public List<String> getUsers() {
+
+        List<String> userList = new ArrayList<>();
+        if (this.isAttributeExist(SCIMConstants.RoleSchemaConstants.USERS)) {
+            MultiValuedAttribute users = (MultiValuedAttribute) this.attributeList
+                    .get(SCIMConstants.RoleSchemaConstants.USERS);
+            List<Attribute> subValuesList = users.getAttributeValues();
+            for (Attribute subValue : subValuesList) {
+                ComplexAttribute complexAttribute = (ComplexAttribute) subValue;
+                Map<String, Attribute> subAttributesList = complexAttribute.getSubAttributesList();
+                if (subAttributesList != null && subAttributesList
+                        .containsKey(SCIMConstants.CommonSchemaConstants.VALUE)) {
+                    userList.add((String) ((SimpleAttribute) (subAttributesList
+                            .get(SCIMConstants.CommonSchemaConstants.VALUE))).getValue());
+                }
+            }
+        }
+        return userList;
+    }
+
+    /**
+     * Get the users of the role with their display names.
+     *
+     * @return List of display names.
+     */
+    public List<String> getUsersWithDisplayName() {
+
+        List<String> displayNames = new ArrayList<>();
+        if (this.isAttributeExist(SCIMConstants.RoleSchemaConstants.USERS)) {
+            MultiValuedAttribute users = (MultiValuedAttribute) this.attributeList
+                    .get(SCIMConstants.RoleSchemaConstants.USERS);
+            List<Attribute> values = users.getAttributeValues();
+            if (values != null) {
+                List<Attribute> subValuesList = users.getAttributeValues();
+                for (Attribute subValue : subValuesList) {
+                    ComplexAttribute complexAttribute = (ComplexAttribute) subValue;
+                    Map<String, Attribute> subAttributesList = complexAttribute.getSubAttributesList();
+                    if (subAttributesList != null && subAttributesList
+                            .containsKey(SCIMConstants.CommonSchemaConstants.DISPLAY)) {
+                        displayNames.add((String) ((SimpleAttribute) (subAttributesList
+                                .get(SCIMConstants.CommonSchemaConstants.DISPLAY))).getValue());
+                    }
+
+                }
+                return displayNames;
+            }
+        }
+
+        return displayNames;
```

Comment:
Let's avoid methods getting expanded to the right. let's return from the reverse if conditions.

Reply:
fixed in [ca64654](https://github.com/wso2/charon/pull/392/commits/ca64654c3f9c03eb95c521cc2f548d9206cbdb4e)fixed in [ca64654](https://github.com/wso2/charon/pull/392/commits/ca64654c3f9c03eb95c521cc2f548d9206cbdb4e)